### PR TITLE
Bump Elasticsearch to 6.5.4

### LINKED ISSUE #383: Bump Elasticsearch dependency to 6.5

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**
- Bump Elasticsearch dependency to 6.5.1
- Adapt Gwithin integration tests to the new geo indexation of ES
- Refactor some code in ARLAS to adapt to the new Java client of ES : 
  - `PutMappingResponse` class no longer exists in 6.5 of Elasticsearch
  - `org.elasticsearch.common.joda.Joda.forPattern()` no longer returns a `org.joda.time.format.DateTimeFormatter` starting of v6.6 of Elasticsearch

Fix #383

### CODE REVIEW

File: arlas-core/src/main/java/io/arlas/server/utils/TimestampTypeMapper.java [hunk 1]
```diff
@@ -57,6 +63,24 @@ else if (formatList.contains(TimestampType.epoch_second.name()))
         return timestamp;
     }
 
+    public static Optional<DateTimeFormatter> getDateTimeFormatter(String format) throws ArlasException {
+        DateTimeFormatter dtf = null;
+        if (!StringUtil.isNullOrEmpty(format)) {
+            TimestampType type = TimestampType.getElasticsearchPatternName(format);
+            if (type.name().equals("UNKNOWN")) {
```
Comment (laurent-thiebaud-gisaia) [inline arlas-core/src/main/java/io/arlas/server/utils/TimestampTypeMapper.java]: Should that be declared as constant?

File: arlas-core/src/main/java/io/arlas/server/utils/TimestampTypeMapper.java [hunk 2]
```diff
@@ -84,19 +108,145 @@ public static void formatDate(Object timestamp, String elasticsearchDateFormat)
...
-        DateTimeFormatter dtf = null;
-        TimestampType type = TimestampType.getElasticsearchPatternName(format);
-        if (type.name().equals("UNKNOWN")) {
-            dtf = DateTimeFormat.forPattern(format);
-        } else {
-            dtf = type.dateTimeFormatter;
-        }
+    private static Long parseDate(String format, String timestampPath) throws ArlasException {
+        DateTimeFormatter dtf = getDateTimeFormatter(format).get();
         if (dtf != null) {
```
Comment (laurent-thiebaud-gisaia) [inline arlas-core/src/main/java/io/arlas/server/utils/TimestampTypeMapper.java]: Why a null check whereas you received an optional(that you get with no test)?

File: arlas-core/src/main/java/io/arlas/server/utils/TimestampTypeMapper.java (hunk 1 above)
Comment (MohamedHamouGisaia) [reply arlas-core/src/main/java/io/arlas/server/utils/TimestampTypeMapper.java]: Done

**Comment by MohamedHamouGisaia** [comment]
@sfalquier I use now a ARLAS tooling class to wrap the ES code

**Pull request merged**